packages/react/tooltip/src/tooltipStateChart.tsx

```diff
@@ -71,7 +65,8 @@ const tooltipStateChart: TooltipStateChart = {
         (event, context, send) => {
           restTimerId = window.setTimeout(
             () => send({ type: 'REST_TIMER_ELAPSE', id: event.id }),
-            REST_THRESHOLD_DURATION
+            // @ts-ignore
+            event.restDuration
```

Comment:
I don't know how to restrict the type further here so it knows which event it is…

Reply:
I've been thinking about this and realising I wasn't fully understanding things when we spoke on the call. I'm now wondering why we need to pass the durations through every `MOUSE_ENTER` or `MOUSE_MOVE` event.

I would usually expect something like:

```jsx
const { restDuration, bypassRestDuration, ...restProps } = props;

React.useEffect(() => {
  send({ type: 'UPDATE_REST_DURATION', restDuration });
}, [send, restDuration]);

React.useEffect(() => {
  send({ type: 'UPDATE_BYPASS_REST_DURATION', bypassRestDuration });
}, [send, bypassRestDuration]);
```

Then the machine would have the following (these are a good example of events that don't transition):

```jsx
on: {
  UPDATE_REST_DURATION: { actions: [assignRestDuration] },
  UPDATE_BYPASS_REST_DURATION: { actions: [assignBypassRestDuration] },
}
```

And your `waitingForRest` would do:


```jsx
entry: [(_, context, send) => {
  restTimerId = window.setTimeout(() => send(), context.restDuration);
}]
```

This would alleviate type issues I believe.

Reply:
It seems strange to have events just to update the context though I guess.

But also I don't think that would work, because if we did that, every single instance of tooltip would override each other because there's one shared state machine for all tooltips.

We do want these timings to change on the fly based on the tooltip currently being used.

Reply:
> It seems strange to have events just to update the context though I guess.

The event could be `PROP_CHANGE`, or whatever you wanna call it. State machine events don't have to map to DOM events, they're just "things that happened".

> We do want these timings to change on the fly based on the tooltip currently being used.

Ah of course! That was the missing piece of the puzzle, cheers. Agree we should leave as is in that case 🙂